substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfiguration.java

```diff
@@ -89,6 +90,11 @@ public void registerWithTargetConstructorClass(ConfigurationCondition condition,
         serializations.add(createConfigurationType(condition, className, customTargetConstructorClassName));
     }
 
+    @Override
+    public void registerLambdaCapturingClass(ConfigurationCondition condition, String lambdaCapturingClassName) {
+        serializations.add(createLambdaCapturingClassConfigurationType(condition, lambdaCapturingClassName.split(LambdaUtils.LAMBDA_SPLIT_PATTERN)[0]));
```

Comment:
Capturing class is the class that captures lambda. And it can be retrieved using the lambda name since name is being created as following `capturingClass$$Lambda$AtomicCounterHashValue`. Names are inconsistent between runs, that is right, and we register only capturing class needed for deserialization of lambda. Lambdas are deserialized using the $deserializeLambda$ method from it's capturing class so we need a name of that class. With unstable names, we cannot register only lambdas we need to be serialized, but all the serializable lambdas from that class. Explanation of this is also added to code as a comment.